src/util/ssl_utils.js

```diff
@@ -71,40 +94,19 @@ function get_ssl_certificate(service) {
         }
 
         cert_info.is_loaded = true;
-        return cert_info.cert;
-    });
-}
-
-// For each cert that was loaded into memory we check if the cert was changed on disk.
-// If so we update it. If any of the certs was updated we return true else we return false.
-async function update_certs_from_disk() {
-    const promiseList = Object.values(certs).map(cert_info =>
-        cert_info.sem.surround(async () => {
-            if (!cert_info.is_loaded) {
-                return false;
-            }
 
-            try {
-                const cert_on_disk = await _read_ssl_certificate(cert_info.dir);
-                if (cert_info.cert.key === cert_on_disk.key) {
-                    return false;
-                }
-
-                cert_info.cert = cert_on_disk;
-                cert_info.is_generated = false;
-                return true;
-
-            } catch (err) {
-                if (err.code !== 'ENOENT') {
-                    dbg.warn(`SSL certificate failed to update from dir ${cert_info.dir}:`, err.message);
-                }
-                return false;
+        try {
+            fs.watch(cert_info.dir, {}, cert_info.file_notification.bind(cert_info));
+        } catch (err) {
+            if (err.code === 'ENOENT') {
+                dbg.warn("Certificate folder ", cert_info.dir, " does not exist. New certificate won't be loaded.");
```

Comment:
another reason to maybe check for the directory in advance and to handle this error there. although not sure we have this split, the level of the print?

Reply:
I don't understand.
This is the only place non-existing directory is handled.

If you think there's still a problem despite what I wrote in previous comment (https://github.com/noobaa/noobaa-core/pull/7502#discussion_r1356202750), let me know.